docs/usage/pydantic_types.md

```diff
@@ -0,0 +1,169 @@
+---
+description: Useful types provided by Pydantic.
```

Comment:
I guess you're aware but all of the types in here should be moved into the relevant file.